### PR TITLE
MAT-10151 Adding ReviewStatus and refactored CqlLibraryService by int…

### PR DESCRIPTION
**Pull request by RohitKandimalla**
…roducing CqlLibraryAccessControlService

## MADiE PR

Jira Ticket: [MAT-10151](https://jira.cms.gov/browse/MAT-10151)
(Optional) Related Tickets:

### Summary
1. Updated "UpdateCqlLibrary" workflow to add review specific actionLog.
2. Refactored CqlLibraryService because it has grown in size and the Checkstyle was complaining about the exceeding number of lines

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if (!isOwnerOrHasAnyRole(cqlLibrary.getLibrarySet(), username, List.of(RoleEnum.SHARED_WITH))) {
+      log.error(
+          "User [{}] does not have permission to modify CQL Library with id [{}]",
+          username,
+          cqlLibrary.getId());
+      throw new PermissionDeniedException("CQL Library", cqlLibrary.getId(), username);
+    }
+  }
+
+  public void checkOwnership(CqlLibrary cqlLibrary, String username) {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java:38]: looks same as AuthUtils.checkOwnership

Comment (adongare) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java:38]: are you planning to remove AuthUtils?

File: src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryReviewService.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        cqlLibraryReviewRepository
+            .findByLibraryId(libraryId)
+            .orElseThrow(() -> new ResourceNotFoundException("CQL Library Review", libraryId));
+
+    final ReviewStatus previousStatus = existing.getStatus();
+    final ReviewStatus newStatus = review.getStatus();
+
+    existing.setStatus(newStatus);
+    existing.setComment(review.getComment());
+    if (review.getLibrarySetId() != null) {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryReviewService.java]: https://github.com/MeasureAuthoringTool/measure-service/pull/1139#discussion_r3631833573

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryReviewService.java]: Resolved

File: src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java:38]: So I created this Service because CqlLibraryService is too long, I copied all the reusable methods from CqlLibraryService into a new service class. Didn't realize we already do have AuthUtils.

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryAccessControlService.java:38]: Yes I am leaning towards replacing Utils, because right now AuthUtils is not dependent on any other services, but CqlLibraryAccessControlService requires LibrarySetService and UserServiceClient. Which is probably not a good idea to be added as dependent when the Utils is not even a spring annotated class.

**Review by adongare — APPROVED** [review]
